### DISCUSSION TITLE
Fast bed level function

### DISCUSSION THREAD

**Discussion by nonnos** [mriscoc/Ender3V2S1#904] [Ideas]
I'm using BLTouch on an Ender3v2 and a bed level mesh of 4x4 points. Often i want to manually adjust the bed level on my Ender 3v2 and then basically only the four corner points are interesting. It would be really neat if there was an "Manual bed-level adjust" mode where the BLTouch only measures the corners (four points instead of 16)  and present the values, allows for adjustment and "re-measure" or "quit".

**Comment by TheWebMachine** [comment]
You want Mesh Tilt Adjustment and it's available in UBL versions of the firmware. *IF* you are on a UBL version firmware, your start gcode would look like this to utilize Mesh Tilt Adjustment:

````
G28 ; Home all axes
G29 A ; activate leveling
G29 J3 ; Tilt mesh
````

You can make whatever manual tramming adjustments you need to and then, when you start your print, the printer will check that tilt change and adjust the mesh accordingly.

**Reply by mriscoc** [reply]
I do not recommend to use the tilt feature, a good bed tramming is enough.
Use:
```gcode
G28  ; Home axes
G29 L0  ; Load a valid mesh into memory from slot 0 (Only if a successful mesh was previously saved there)
G29 A  ; Activate the leveling system if a valid mesh is in memory
```

**Reply by TheWebMachine** [reply]
I know some folks had problems with it, but on my F4 it worked flawlessly when I was running the UBL firmware. I also know Marlin has put put a few new fixes for tilt in recent months, so those who had issues before may find those issues resolved in your latest version.

**Comment by mriscoc** [comment]
> i want to manually adjust the bed level on my Ender 3v2

Yes it is possible going to the Advanced/Mesh Leveling/Edit Mesh

![mesh-leveling-3](https://user-images.githubusercontent.com/2745567/236892313-b1c42d97-4eea-4bc2-8772-9550bd02920c.jpg)
![edit-mesh](https://user-images.githubusercontent.com/2745567/236892393-4442807a-2f12-4fb1-9c20-9672018d45c8.jpg)

**Reply by nonnos** [reply]
I tested the ”Tilt Mesh” but what I want is a fast way to run a 2x2 bed level mesh showing corner values for manual trim (and re-run) of the bed level and when printing using 5x5 mesh.

**Reply by TheWebMachine** [reply]
Unfortunately, that still involves the manual step of using the Tramming Wizard or Manual Tram to hit those 4 corners for manual adjustment. That's what Tilt Mesh is suppose to preclude. It checks 3-9 points (J[1-3]) and makes minor adjustment to the existing mesh to match the slight deviation tram/bed plane that easily happens with the roller/rail design of the Y-axis of the machine.

Tramming, to me, is at best an approximation to keep bed leveling values within measurable tolerance. With the huge bed on my Plus, I can tram perfectly, then run the bed back and forth a dozen times, only to find the tram has slipped on a couple corners by as much as 0.05 because rollers drift and warble slightly, no matter how you have them tightened (too tight and they wear down and loosen again).

That is the premise of Tilt Mesh:
* You do you manual tramming occasionally to account for roller wear and bring the bed to general level state
* You redo mesh only as needed, like when changing magnetic bed plates, or occasionally when you re-tram to account for bed surface wear/warp over time
* You incorporate Tilt Mesh into start code, which checks for and corrects the plane of the mesh to match those slight variations that come up in the tram over time

The mesh only needs redone if the actual physical surface of the bed changes, like changing the bed plate or after many prints where the heat/cool cycle might cause slight warping or wear in the case of PEI coated beds.

Like I said, I know Tramming used to be hit or miss, but it has been working well [for me] in the latest Marlin versions and has cut my print start times to 1/3 what they used to be when I was redoing mesh all the time. As a perfectionist, I found constantly tramming to be a fool's errand...these printers were just not designed to be *that* precise.

(Note: *Was* working for me, as I am now on CVLM FW that doesn't have UBL/Tilt Mesh due to space constaints on the F4.)
